### ISSUE TITLE
Possibility to arrange slides on a grid as in reveal.js

### ISSUE THREAD

**Issue by gevero** [slidevjs/slidev#1187]
**Sequential arrangement of slides**
Right now the slides are simply arranged sequentially

**Slides arranged on a grid**
It would be nice to have slides arranged on a grid, as implemented in reveal.js. It would be very useful when teaching classes. With slides on a grid, the first slide of a column would be the one with the teaching content, and the ones below could be used as a virtual whiteboard.

**Possible alternative**
A possible alternative would be to have a whiteboard plugin. With the possibility to pull up a whiteboard associated with each slides.

**Comment by piranna** [comment]
I like the idea, but one limitation I found in Reveal.js grid is that you can only append them down or right, since all slides are arranged in a 2D space. I would like to be able to add them to the left too, also if they would "overlap" in a 2D space, and maybe move to anywhere else when doing a transition, not only to the close slides.

**Comment by Marco-Christiani** [comment]
How should this be implemented, as a vue component?

**Comment by heyalexnet** [comment]
I use this feature to design presentations where I can go into more details only if needed. The top slides are the main content and if I get questions or feel I need to go deeper then I move down.

**Comment by sultaniman** [comment]
That would be good to have.

**Comment by psifertex** [comment]
I'd very much like this as well, this is my favorite design decision from reveal.js and it saddened me to see it not implemented in slidev.

To clarify though for my use-case, a whiteboard plugin does not help at all. The value for me is being able to organize topics logically into "chapters" such that every section or chapter starts in a new column. This helps not only when creating slides because I can lay out my thoughts better, but because I can more easily organize/move around slides by topics. 

Additionally, while presenting it's much faster for me to navigate to any other specific slide because I first identify the column by scrolling horizontally, and then I scroll vertically within that column to the specific slide. Less clicks to get to where I need to which is super important when presenting live!

**Comment by jpgehrig** [comment]
Count my vote for this feature!

My use case involves incorporating additional content "under" a slide. I use the same deck to present to different audiences. When addressing business people, I stay at the top level and do not need to skip slides. When presenting to a technical audience, I include the additional content.

The same presentation, adapted to the audience.

Thanks for slidev, love it !

**Comment by encelo** [comment]
I also vote for this feature!
It's the only thing that I miss from reveal.js and I think it's very useful, as it has already been noted, for bonus slides or to organize the flow of a presentation.

**Comment by JPtenBerge** [comment]
Yep, vertical slides would be great. My company gives a lot of courses, we make our material in Markdown and have our own presentation CLI to make a presentation out of it. We organize the chapters/modules of our course the same way @psifertex explained.

Would be great if Slidev got it. Reveal.js is getting less appealing every day. I'm hoping we can switch to Slidev.

**Comment by wbxl2000** [comment]
This feature might need big changes to how Slidev handles slide switching. It would be great if someone could share real slide examples as use cases :)

And here's jsut a brainstorming idea (still very raw)

syntax example:

```markdown
---
src: ./pages/imported-slides.md
---

# Navigation

Hover on the bottom-left corner to see the navigation's controls panel
```

In current version, this would only show content from `imported-slides.md`. The proposal is to:
1. First display the local content (`# Navigation section`)
2. Then show the imported slides
3. Client NavControls could switch to vertical mode
4. Imported slides would be collapsible under the `# Navigation section` for vertical mode

Benefits:
- Keeps main markdown file clean and structured
- Maintains backward compatibility 
- Future extensibility (e.g. more yaml config like`verticle-default: true`)

**Comment by psifertex** [comment]
Here's a few examples:

https://github.com/psifertex/breaking_decompilers
preview: http://psifertex.github.io/breaking_decompilers

https://github.com/psifertex/a-brief-history-of-ctf
preview: https://psifertex.github.io/a-brief-history-of-ctf/#/

(just press escape to see the 2d overview and see how navigation works)

**Comment by ElykDeer** [comment]
This is a feature I've wanted for a long time. Had AI implement it, seems to work great but I'm not webscale enough to give it a proper review. Sharing it [here](https://github.com/ElykDeer/slidev) if someone else would like to take it up. Diff is fairly small, ~400 lines: https://github.com/slidevjs/slidev/compare/main...ElykDeer:slidev:main

Has docs, tests, and all that. Only note is that metadata between two slides in the same col might be a little weird.

**Comment by watermarkhu** [comment]
@ElykDeer Why don't you create the PR?

**Comment by tiziodcaio** [comment]
I think we should use a YAML-friendly syntax instead of the `--` separator. I tried rebasing the patch onto `main`, but I think the community needs to specify more precisely how we want to implement this feature.

EDIT: in my opinion if we want to recycle the idea we might be use `----`.

**Comment by tiziodcaio** [comment]
@wbxl2000 idea is better! Maybe I can implement it and see if it works

**Comment by ElykDeer** [comment]
Rebased my branch, I'll open a PR.

Changed from -- to ---- so there don't need to be any parser changes.

---- is basically a shortcut for `nested: true`; if you have a normal metadata block with `nested: true` and a `src:` include in there it should nest the whole sub-deck under the slide above it. If you have `nested: true` on just a single slide, it'll nest under the slide above it. Could change it to/add `verticle-default:` from @wbxl2000's suggestion if that's strongly desired.
